pkg/meta/model/job.go

```diff
@@ -546,6 +546,9 @@ func (job *Job) Encode(updateRawArgs bool) ([]byte, error) {
 // decode special args for this job.
 func (job *Job) Decode(b []byte) error {
 	err := json.Unmarshal(b, job)
+	if job.MayNeedReorg() && job.ReorgMeta == nil {
```

Comment:
`MayNeedReorg` depends on a context var set at DDL executor, when job worker run it, it's not there

maybe move it to where `job.ReorgMeta` is used

Reply:
update in https://github.com/pingcap/tidb/pull/57611/commits/a128426b9b223d1199c87d1875d43b341dee02e7 PTAL again